### PR TITLE
Change OSB server lib with reverse proxy

### PR DESCRIPTION
**Pull request by pankrator**
Depends on https://github.com/Peripli/service-manager/pull/113

### CODE REVIEW

File: pkg/sm/config.go [hunk 1]
```diff
@@ -32,20 +32,21 @@ func NewConfig(env env.Environment) (*Config, error) {
...
-type Config struct {
+// Settings type holds SM Client config properties
+type Settings struct {
 	User              string
 	Password          string
 	Host              string
-	OsbAPI            string
-	RequestTimeout    time.Duration
-	SkipSslValidation bool
+	OsbAPI            string        `mapstructure:"osb_api"`
```
Comment (dotchev) [inline pkg/sm/config.go]: `OSBAPI`, see https://github.com/golang/go/wiki/CodeReviewComments#initialisms
What is this URL, path, version? Make it clear in the name.

File: pkg/sm/config.go [hunk 2]
```diff
@@ -32,20 +32,21 @@ func NewConfig(env env.Environment) (*Config, error) {
...
+	RequestTimeout    time.Duration `mapstructure:"request_timeout"`
+	ResyncPeriod      time.Duration `mapstructure:"resync_period"`
+	SkipSslValidation bool          `mapstructure:"skip_ssl_validation"`
```
Comment (dotchev) [inline pkg/sm/config.go]: `SkipSSLValidation`, see https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: pkg/sm/config.go [hunk 3]
```diff
@@ -32,20 +32,21 @@ func NewConfig(env env.Environment) (*Config, error) {
 	return config.Sm, nil
 }
 
-// Config type holds SM Client config properties
-type Config struct {
+// Settings type holds SM Client config properties
+type Settings struct {
 	User              string
 	Password          string
 	Host              string
```
Comment (dotchev) [inline pkg/sm/config.go]: Rename `Host` to `URL` to reflect better the actual meaning

**Review by dotchev — DISMISSED** [review]
Improve names in Settings

**Review by KirilKabakchiev — DISMISSED** [review]

File: Gopkg.toml [hunk 4]
```diff
@@ -53,25 +25,21 @@
...
+ version = "=1.3.0"
 
 [[constraint]]
   name = "github.com/spf13/pflag"
   version = "=1.0.1"
 
 [[constraint]]
   name = "github.com/Peripli/service-manager"
-  branch = "master"
+  branch = "reverse-proxy"
```
Comment (KirilKabakchiev) [inline Gopkg.toml]: change to master and rebuild lock before merging

**Review by KirilKabakchiev — APPROVED** [review]

**Pull request merged**